crates/ruff_linter/src/rules/pydoclint/rules/check_docstring.rs

```diff
@@ -607,25 +647,41 @@ fn parse_parameters_google(content: &str) -> Vec<&str> {
 /// b : int
 ///     The second number to add.
 /// ```
-fn parse_parameters_numpy(content: &str) -> Vec<&str> {
-    let mut entries: Vec<&str> = Vec::new();
+fn parse_parameters_numpy(content: &str, content_start: TextSize) -> Vec<ParameterEntry<'_>> {
+    let mut entries: Vec<ParameterEntry> = Vec::new();
     let mut lines = content.lines();
     let Some(dashes) = lines.next() else {
         return entries;
     };
     let indentation = &dashes[..dashes.len() - dashes.trim_start().len()];
+
+    let mut current_pos = dashes.len() + 1;
     for potential in lines {
+        let line_start = current_pos;
+        let line_end = current_pos + potential.len();
+
         if let Some(entry) = potential.strip_prefix(indentation) {
             if entry
                 .chars()
                 .next()
                 .is_some_and(|first_char| !first_char.is_whitespace())
             {
                 if let Some(param) = entry.split(':').next() {
-                    entries.push(param.trim_end().trim_start_matches('*'));
+                    let param_name = param.trim_end().trim_start_matches('*');
+                    let param_start = line_start + indentation.len();
+                    let param_end = param_start + param.len();
+
+                    entries.push(ParameterEntry {
+                        name: param_name,
+                        range: TextRange::new(
+                            content_start + TextSize::new(param_start as u32),
+                            content_start + TextSize::new(param_end as u32),
```

Comment:
I think we try to avoid `as` casts since they can silently misbehave. This should be safe since the ranges are from within the document, and the document has to be less than `u32::MAX` in length, but let's unwrap in case that assumption ever breaks down:

```suggestion
                            content_start + TextSize::try_from(param_start as u32).unwrap(),
                            content_start + TextSize::try_from(param_end).unwrap(),
```

Ah this is what clippy is complaining about too.